andela/coding-geeks-ah-backend
# #167484608 Users should be able to report an article

### What does this PR do?
- User should be able to report articles that violate terms of agreement or are plagiarised to the authors haven admin
### Description of Task to be done?
- Create a new controller for report functionalities.
- Write test of testing this feature
- Document this feature with swagger
### How should this be manually tested?
- Clone this branch
- Run `npm start`
- Run migration
- Use this route `http://localhost:3000/api/v1/articles/:slug/reports` to report a specific article using its slug.   and use `POST` as http method
- Use this route `http://localhost:3000/api/v1/articles/:slug/reports` to view a reports  of a specific article. and  use `GET` as http method
- Use this route `http://localhost:3000/api/v1/articles/reports/all` to view all reports  and  use `GET` as http method
- Use this route `http://localhost:3000/api/v1/articles/:slug/reports/:reportId` to delete a report  and  use `DELETE` as http method.
`N.B:` Only admins and super-admin are allowed to see the reported articles
### Any background context you want to provide?
- N/A
### What are the relevant pivotal tracker stories?
- [Users should be able to report an article](https://www.pivotaltracker.com/story/show/167484608)
### Questions
- N/A
### Screenshots
#### Report an article
<img width="1138" alt="Screen Shot 2019-08-28 at 16 59 09" src="https://user-images.githubusercontent.com/39116572/63867477-32335800-c9b5-11e9-9d02-c3d0b1044491.png">
#### Get all reports
<img width="1139" alt="Screen Shot 2019-08-28 at 17 00 50" src="https://user-images.githubusercontent.com/39116572/63867592-6dce2200-c9b5-11e9-9c9f-d2ba11fb7423.png">
#### Delete the report
<img width="1126" alt="Screen Shot 2019-08-28 at 17 02 20" src="https://user-images.githubusercontent.com/39116572/63867712-a0781a80-c9b5-11e9-943c-d320eef07d0d.png">
